src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -549,6 +549,10 @@ bool Creature::UpdateEntry(uint32 Entry, const CreatureData* data, bool changele
     SetDetectionDistance(cInfo->detection_range);
 
     LoadSpellTemplateImmunity();
+
+    // We must update last scriptId or it looks like we reloaded a script, breaking some things such as gossip temporarily
+    LastUsedScriptID = GetScriptId();
+
```

Comment:
To me this just looks weird. We don't pass this variable anywhere after it. So why would it make a difference?

Reply:
https://github.com/azerothcore/azerothcore-wotlk/blob/8497a3683023166cab1123751575ca3131ce08a2/src/server/game/Entities/Creature/Creature.cpp#L1045

It is the same meathod that is used on creature create that enables it to load its script.